tests/test_metadata.py

```diff
@@ -12,43 +12,69 @@
 # limitations under the License.
 
-from rapids_metadata.metadata import Metadata
-
-
-def test_inherit():
-    parent = Metadata()
-    parent.packages.update({"package-1", "package-2"})
-    parent.non_alpha_spec_packages.update({"package-1"})
-    parent.non_cuda_suffixed_packages.update({"package-2"})
-
-    metadata = Metadata(parent)
-    assert metadata.packages == {"package-1", "package-2"}
-    assert metadata.non_alpha_spec_packages == {"package-1"}
-    assert metadata.non_cuda_suffixed_packages == {"package-2"}
-    assert parent.packages == {"package-1", "package-2"}
-    assert parent.non_alpha_spec_packages == {"package-1"}
-    assert parent.non_cuda_suffixed_packages == {"package-2"}
-
-    metadata.packages.update({"package-3"})
-    metadata.non_alpha_spec_packages.update({"package-3"})
-    metadata.non_cuda_suffixed_packages.update({"package-3"})
-    assert metadata.packages == {"package-1", "package-2", "package-3"}
-    assert metadata.non_alpha_spec_packages == {"package-1", "package-3"}
-    assert metadata.non_cuda_suffixed_packages == {"package-2", "package-3"}
-    assert parent.packages == {"package-1", "package-2"}
-    assert parent.non_alpha_spec_packages == {"package-1"}
-    assert parent.non_cuda_suffixed_packages == {"package-2"}
-
-
-def test_alpha_spec_packages():
-    metadata = Metadata()
-    metadata.packages.update({"package-1", "package-2"})
-    metadata.non_alpha_spec_packages.update({"package-1"})
-    assert metadata.alpha_spec_packages == {"package-2"}
-
-
-def test_cuda_suffixed_packages():
-    metadata = Metadata()
-    metadata.packages.update({"package-1", "package-2"})
-    metadata.non_cuda_suffixed_packages.update({"package-1"})
-    assert metadata.cuda_suffixed_packages == {"package-2"}
+from unittest.mock import Mock, patch
+
+import pytest
+from packaging.version import InvalidVersion
+from rapids_metadata import metadata as md
+
+
+class TestRAPIDSVersion:
+    @pytest.fixture
+    def metadata(self):
+        return md.RAPIDSVersion(repositories={
+            "repo1": md.RAPIDSRepository(packages={
+                "package1": md.RAPIDSPackage(has_alpha_spec=True, has_cuda_suffix=True),
+                "package2": md.RAPIDSPackage(has_alpha_spec=True, has_cuda_suffix=False),
+            }),
+            "repo2": md.RAPIDSRepository(packages={
+                "package3": md.RAPIDSPackage(has_alpha_spec=False, has_cuda_suffix=True),
+                "package4": md.RAPIDSPackage(has_alpha_spec=False, has_cuda_suffix=False),
+            }),
+        })
+
+    def test_all_packages(self, metadata):
+        assert metadata.all_packages == {"package1", "package2", "package3", "package4"}
+
+    def test_alpha_spec_packages(self, metadata):
+        assert metadata.alpha_spec_packages == {"package1", "package2"}
+
+    def test_cuda_suffixed_packages(self, metadata):
+        assert metadata.cuda_suffixed_packages == {"package1", "package3"}
+
+
+class TestRAPIDSMetadata:
```

Comment:
`pytest` fully supports grouping by classes: https://docs.pytest.org/en/8.2.x/getting-started.html#group-multiple-tests-in-a-class and I think in this case it's appropriate to group these tests, since they're testing different methods of different classes.

Reply:
From my perspective, the fact that these tests cover similar functionality isn't a sufficient reason to deviate from how Python tests are structured across the rest of RAPIDS and use classes instead of functions.

There are other options for grouping tests together that work with test cases as plain functions, like:

* using separate test files
* code comments separating blocks of test functions
* `pytest` markers ([pytest markers docs](https://docs.pytest.org/en/stable/example/markers.html))

I think that if there's not some class-specific functionality you want to take advantage of, then functions should be preferred for consistency with the rest of RAPIDS Python codebases.

But I don't feel strongly enough about this to block the PR over it. Thanks for considering it.

Reply:
I've split them up into separate files.